core/camel-api/src/main/java/org/apache/camel/TimeoutMap.java

```diff
@@ -67,16 +61,6 @@
      */
     V putIfAbsent(K key, V value, long timeoutMillis);
 
-    /**
-     * Callback when the value has been evicted
-     *
-     * @param key the key
-     * @param value the value
-     * @return <tt>true</tt> to remove the evicted value,
-     *         or <tt>false</tt> to veto the eviction and thus keep the value.
-     */
-    boolean onEviction(K key, V value);
```

Comment:
`purge` and `onEviction` are specific to `DefaultTimeoutMap` and were never accessed via the interface